src/submodules/pathfinding/astar_continuous.jl

```diff
@@ -0,0 +1,199 @@
+Calculate and store the shortest path to move the agent from its current position to
+`target` (a continuous position e.g. `(1.2, 5.7)`) using the provided `pathfinder`.
+
+For pathfinding in models with [`ContinuousSpace`](@ref).
+
+Use this method in conjuction with [`move_along_route!`](@ref).
+"""
+function set_target!(
+    agent::A,
+    target::NTuple{D,Float64},
+    pathfinder::AStar{D},
+    model::ABM{<:ContinuousSpace{D},A},
+) where {D,A<:AbstractAgent}
+    path = find_continuous_path(pathfinder, agent.pos, target, model)
+    isnothing(path) && return
+    pathfinder.agent_paths[agent.id] = path
+end
+
+"""
+    Pathfinding.set_best_target!(agent, targets::Vector{NTuple{D,Float64}}, pathfinder::AStar{D}, model::ABM{<:ContinuousSpace{D}})
+Calculate and store the best path to move the agent from its current position to
+a chosen target position taken from `targets` for models using the provided `pathfinder`.
+
+For pathfinding in models with [`ContinuousSpace`](@ref).
+
+The `condition = :shortest` keyword retuns the path which is shortest
+(allowing for the conditions of the pathfinder) out of the possible target
+positions. Alternatively, the `:longest` path may also be requested.
+
+Returns the position of the chosen target.
+"""
+function set_best_target!(
+    agent::A,
+    targets::Vector{NTuple{D,Float64}},
+    pathfinder::AStar{D},
+    model::ABM{<:ContinuousSpace{D}};
+    condition::Symbol = :shortest,
+) where {D,A<:AbstractAgent}
+    @assert condition ∈ (:shortest, :longest)
+    compare = condition == :shortest ? (a, b) -> a < b : (a, b) -> a > b
+    best_path = Path{D,Float64}()
+    best_target = nothing
+    for target in targets
+        path = find_continuous_path(pathfinder, agent.pos, target, model)
+        isnothing(path) && continue
+        if isempty(best_path) || compare(length(path), length(best_path))
+            best_path = path
+            best_target = target
+        end
+    end
+
+    isnothing(best_target) && return
+    pathfinder.agent_paths[agent.id] = best_path
+    return best_target
+end
+
+"""
+    move_along_route!(agent, model::ABM{<:ContinuousSpace{D}}, pathfinder::AStar{D}, speed, dt = 1.0)
+Move `agent` for one step along the route toward its target set by
+[`Pathfinding.set_target!`](@ref) at the given `speed` and timestep `dt`.
+
+For pathfinding in models with [`ContinuousSpace`](@ref)
+
+If the agent does not have a precalculated path or the path is empty, it remains stationary.
+"""
+function Agents.move_along_route!(
+    agent::A,
+    model::ABM{<:ContinuousSpace{D},A},
+    pathfinder::AStar{D},
+    speed::Float64,
+    dt::Real = 1.0,
+) where {D,A<:AbstractAgent}
+    isempty(agent.id, pathfinder) && return
+    from = agent.pos
+    next_pos = agent.pos
+    while true
+        next_waypoint = first(pathfinder.agent_paths[agent.id])
+        dir = get_direction(from, next_waypoint, model)
+        dist_to_target = √sum(dir .^ 2)
```

Comment:
We have LinearAlgebra [imported](https://github.com/JuliaDynamics/Agents.jl/blob/b8c014db23551eb3a3519ffc7f6d03f1911988cc/src/Agents.jl#L12), there's little reason not to use it herein.